File: 정세희/hash1.js
function solution(participant, completion) {
  var answer = '';
  let m = new Map();
  for(let i in completion) {
      if(m.has(completion[i]))
          m.set(completion[i], m.get(completion[i])+1);
      else
          m.set(completion[i], 1);
  }
  for(let i in participant) {
      if(!m.has(participant[i]) || m.get(participant[i]) === 0)
          answer = participant[i];
      else
          m.set(participant[i], m.get(participant[i])-1);
  }
  return answer;
}

File: 정세희/hash1.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    string answer = "";
    unordered_map<string, int> map;
    for(auto player: participant) {
        if(map.end() == map.find(player))
            map.insert(make_pair(player,1));
        else
            map[player]++;
    }

    for(auto player: completion) {
        map[player]--;
    }
    for(auto player: participant) {
        if(map[player] != 0) {
            answer = player;
            break;
        }
    }
    return answer;
}